code/modules/reagents/chemistry/machinery/chem_synthesizer.dm

```diff
@@ -0,0 +1,85 @@
+/obj/machinery/chem_dispenser/chem_synthesizer //formerly SCP-294 made by mrty, but now only for testing purposes
+	name = "\improper debug chemical synthesizer"
+	desc = "If you see this, yell at adminbus."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "dispenser"
+	amount = 10
+	resistance_flags = INDESTRUCTIBLE | FIRE_PROOF | ACID_PROOF | LAVA_PROOF
+	working_state = null
+	nopower_state = null
+	flags_1 = NODECONSTRUCT_1
+	var/static/list/shortcuts = list(
+		"meth" = "methamphetamine",
+		"tricord" = "tricordrazine"
+	)
+	var/mutable_appearance/top_overlay
+
+/obj/machinery/chem_dispenser/chem_synthesizer/Initialize()
+	. = ..()
+	GLOB.poi_list += src
+	top_overlay = mutable_appearance(icon, "disp_beaker", layer = ABOVE_ALL_MOB_LAYER)
+	update_icon()
+
+/obj/machinery/chem_dispenser/chem_synthesizer/update_icon()
+	cut_overlays()
+	add_overlay(top_overlay)
+
+/obj/machinery/chem_dispenser/chem_synthesizer/Destroy()
+	. = ..()
+	GLOB.poi_list -= src
+	QDEL_NULL(top_overlay)
+
+/obj/machinery/chem_dispenser/chem_synthesizer/display_beaker()
+	return
+
+/obj/machinery/chem_dispenser/chem_synthesizer/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, \
+											datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "chem_synthesizer", name, 390, 315, master_ui, state)
+		ui.open()
+
+/obj/machinery/chem_dispenser/chem_synthesizer/ui_act(action, params)
+	if(..())
+		return
+	update_icon()
+	switch(action)
+		if("ejectBeaker")
+			if(beaker)
+				beaker.forceMove(drop_location())
+				if(Adjacent(usr) && !issilicon(usr))
+					usr.put_in_hands(beaker)
+				beaker = null
+				. = TRUE
+		if("input")
+			var/input_reagent = replacetext(lowertext(input("Enter the name of any liquid", "Input") as text), " ", "") //95% of the time, the reagent id is a lowercase/no spaces version of the name
+			if(shortcuts[input_reagent])
+				input_reagent = shortcuts[input_reagent]
+			else
+				input_reagent = find_reagent(input_reagent)
+			if(!input_reagent || !GLOB.chemical_reagents_list[input_reagent])
+				say("OUT OF RANGE")
+				return
+			else
+				if(!beaker)
+					return
+				else if(!beaker.reagents && !QDELETED(beaker))
+					beaker.create_reagents(beaker.volume)
+				beaker.reagents.add_reagent(input_reagent, amount)
```

Comment:
Always adding only ten units seems a little inflexible, maybe make it adjustable?

Reply:
Good idea, let me try to mess around with TGUI.

Reply:
Alright this will take a while

Reply:
I've managed to get TGUI to work, but entering a value doesn't actually change the chem synthesizer's `amount` var. Does anyone have an idea?

Reply:
Take a look at what the ORM does, roughly like:

```html
<input value={{amount}} placeholder='###' class="number"/>
<ui-button class='center' action='dispense' state={{(amount >= 1) ? null : 'disabled'}} params='{ "amount" : {{amount}} }' >
```

Maybe use it for the chem ID too if you can pull it off? Just a suggestion.

Reply:
Thanks for the suggestion, I'll give it a shot.

Reply:
@SpaceManiac I realized that I need a whole bunch of practice until I can make it work. If you're fine with merging the PR as it is right now, I'll figure out a way to implement changing the amount eventually.

Reply:
It's VVable if nothing else.